plat/kvm/efi.c

```diff
@@ -76,30 +80,46 @@ static __sz utf16_to_ascii(const char *str16, char *str, __sz max_len)
 	return i + 1;
 }
 
-static void _uk_efi_crash(void)
+static void uk_efi_do_crash(void)
 {
 	const char reset_data[] = "UK EFI SYSTEM CRASH";
 
 	uk_efi_rs->reset_system(UK_EFI_RESET_SHUTDOWN, UK_EFI_SUCCESS,
 				sizeof(reset_data), (void *)reset_data);
 }
 
-#ifdef CONFIG_KVM_BOOT_PROTO_EFI_STUB_DEBUG
-#define UK_EFI_MAX_CRASH_STR_LEN				256
+#if CONFIG_LIBUKDEBUG_PRINTD
+#define UK_EFI_MAX_FMT_STR_LEN					256
+
+static void uk_efi_printf(const char *str, ...)
+{
+	char fmt_str[UK_EFI_MAX_FMT_STR_LEN];
+	char str_tmp[UK_EFI_MAX_FMT_STR_LEN];
+	__s16 str16[UK_EFI_MAX_FMT_STR_LEN];
+	va_list ap;
+
+	sprintf(fmt_str, "dbg: [%s] <%s @ %4u> %s\r", uk_libname_self(),
+		STRINGIFY(__BASENAME__), __LINE__, str);
+
+	va_start(ap, str);
+	vsprintf(str_tmp, fmt_str, ap);
+	va_end(ap);
+
+	ascii_to_utf16(str_tmp, (char *)str16, UK_EFI_MAX_FMT_STR_LEN - 1);
+	uk_efi_st->con_out->output_string(uk_efi_st->con_out, str16);
+}
+
+#define uk_efi_pr_debug uk_efi_printf
 /* UEFI for proper \n, we must also use CRLF */
-#define uk_efi_crash(str)						\
-	do {								\
-		__s16 str16[UK_EFI_MAX_CRASH_STR_LEN];			\
-									\
-		ascii_to_utf16("[uk_efi]: "str"\r", (char *)str16,	\
-			       UK_EFI_MAX_CRASH_STR_LEN - 1);		\
-		uk_efi_st->con_out->output_string(uk_efi_st->con_out,	\
-						  str16);		\
-		_uk_efi_crash();					\
+#define UK_EFI_CRASH(...)					\
+	do {							\
+		uk_efi_printf(__VA_ARGS__);			\
+		uk_efi_do_crash();				\
 	} while (0)
 #else
-#define uk_efi_crash(str)					_uk_efi_crash()
-#endif
+#define uk_efi_pr_debug(...)
+#define UK_EFI_CRASH(str)					uk_efi_do_crash()
```

Comment:
Hmm this line seems to be over 80 characters, maybe let's have one tab less :thinking:.